libs/resolver/emit.rs

```diff
@@ -410,18 +511,11 @@ impl<TInNpmPackageChecker: InNpmPackageChecker, TSys: EmitterSys>
 
 #[allow(clippy::result_large_err)]
 fn transpile(
-  parsed_source_cache: &ParsedSourceCache,
-  specifier: &Url,
-  media_type: MediaType,
+  parsed_source: ParsedSource,
   module_kind: deno_ast::ModuleKind,
-  source: ArcStr,
   transpile_options: &deno_ast::TranspileOptions,
   emit_options: &deno_ast::EmitOptions,
 ) -> Result<EmittedSourceText, EmitParsedSourceHelperError> {
-  // nothing else needs the parsed source at this point, so remove from
-  // the cache in order to not transpile owned
-  let parsed_source = parsed_source_cache
-    .remove_or_parse_module(specifier, source, media_type)?;
```

Comment:
Again, better to do this at a higher level (which we already do).

Reply:
Oh, it looks like we do this in order to not clone when transpiling. I'll have to revert some of this.